plugins/exporter.koplugin/main.lua

```diff
@@ -460,265 +120,137 @@ function Exporter:updateMyClippings(clippings, new_clippings)
-    -- Use wide_space to avoid crengine to treat it specially.
-    local wide_space = "\227\128\128"
-    local file = io.open(self.text_clipping_file, "a")
-    if file then
-        file:write(title .. "\n" .. wide_space .. "\n")
-        for _ignore1, chapter in ipairs(booknotes) do
-            if chapter.title then
-                file:write(wide_space .. chapter.title .. "\n" .. wide_space .. "\n")
-            end
-            for _ignore2, clipping in ipairs(chapter) do
-                file:write(wide_space .. wide_space ..
-                            T(_("-- Page: %1, added on %2\n"),
-                                clipping.page, os.date("%c", clipping.time)))
-                if clipping.text then
-                    file:write(clipping.text)
-                end
-                if clipping.note then
-                    file:write("\n---\n" .. clipping.note)
-                end
-                if clipping.image then
-                    file:write(_("<An image>"))
-                end
-                file:write("\n-=-=-=-=-=-\n")
-            end
-        end
-
-        file:write("\n")
-        file:close()
-    end
+function Exporter:getAllNotes()
+    local clippings = self.config:readSetting("clippings") or {}
+    clippings = self:updateHistoryClippings(clippings, self.parser:parseHistory())
+    clippings = self:updateMyClippings(clippings, self.parser:parseMyClippings())
+    self.config:saveSetting("clippings", clippings)
+    self.config:flush()
+    return clippings
 end
 
-function Exporter:exportBooknotesToJoplin(client, title, booknotes)
-    if not client:ping() then
-        error("Cannot reach Joplin server")
-    end
-
-    local note_guid = client:findNoteByTitle(title, self.joplin_notebook_guid)
-    local note = ""
-    for _, chapter in ipairs(booknotes) do
-        if chapter.title then
-            note = note .. "\n\t*" .. chapter.title .. "*\n\n * * *"
-        end
-
-        for _, clipping in ipairs(chapter) do
-            note = note .. os.date("%Y-%m-%d %H:%M:%S \n", clipping.time)
-            note = note .. clipping.text
-            if clipping.note then
-                note = note .. "\n---\n" .. clipping.note
-            end
-            note = note .. "\n * * *\n"
-        end
-    end
-
-    if note_guid then
-        client:updateNote(note_guid, note)
-    else
-        client:createNote(title, note, self.joplin_notebook_guid)
+function Exporter:addToMainMenu(menu_items)
+    local submenu = {}
+    for k, v in pairs(self.targets) do
+        submenu[#submenu + 1] = v:getMenuTable()
     end
+    table.sort(submenu, function(v1, v2)
+        return v1.text < v2.text
+    end)
 
-end
-
-function Exporter:exportBooknotesToReadwise(client, title, booknotes)
-    client:createHighlights(booknotes)
+    menu_items.exporter = {
+        text = _("Export highlights"),
+        sub_item_table = {
+            {
+                text = _("Export all notes in this book"),
```

Comment:
Were these kept the same, to what extent should it be reviewed?

Reply:
Yup, it's almost the same, except checkboxes for different services are now under "Choose formats and services" instead of the main menu of the plugin

Reply:
Sounds good. :+1: